t/28-signalblocker.t

```diff
@@ -48,7 +48,7 @@ $SIG{CHLD} = sub { $received_sigchld += 1; note "received SIGCHLD $received_sigc
     # make the number of threads to spawn configurable
     my $thread_count = tinycv::default_thread_count();
     my $thread_count_for_testing = $ENV{OS_AUTOINST_TEST_THREAD_COUNT} || $thread_count;
-    diag "threads used: $thread_count_for_testing of $thread_count";
+    note "threads used: $thread_count_for_testing of $thread_count";
```

Comment:
ok, Martchus explained me that he sees this line as really necessary because tests could run into timeouts and then it's unclear what the thread count was. We don't use this approach anywhere else but we discussed different options how to do better. We talked about saving the output into a debug log file and outputting that in a `END { }` hook but this won't ever be called if we have a deadlock. I could simulate a problem with

```diff
diff --git a/ppmclibs/tinycv_impl.cc b/ppmclibs/tinycv_impl.cc
index ac1f354e..c8386975 100644
--- a/ppmclibs/tinycv_impl.cc
+++ b/ppmclibs/tinycv_impl.cc
@@ -360,10 +360,10 @@ void create_opencv_threads(int thread_count)
     std::mutex m;
     std::condition_variable cv;
     int threads_spawned = 0;
-    parallel_for_(Range(0, thread_count), RunFunctionInParallel([&] (const Range &) {
+    parallel_for_(Range(0, 100), RunFunctionInParallel([&] (const Range &) {
         // keep the thread idling until the expected number of threads has been spawned
         std::unique_lock<std::mutex> lock(m);
-        if (++threads_spawned >= thread_count) {
+        if (++threads_spawned >= 100) {
             cv.notify_all(); // note: The last thread releases all previously spawned threads.
         } else {
             cv.wait(lock);
```

and `make && prove -I. -v t/28-signalblocker.t` which makes t/28-signalblocker.t block indefinitely, and not be aborted by OpenQA::Test::TimeLimit . So our next suggestion is to implement a simple testing specific alarm signal handler in ppmclibs/tinycv_impl.cc that would output such debug information.

Reply:
We could write to a debug logfile and upload it as an artififact in a later step, right?

Reply:
This is also about OBS runs. (In fact, if such a problem would ever occur again we'd likely see it again on OBS first.)

Reply:
I added a commit to enable verbose test output in OBS. I think we had this originally but lost it over time. This should help to address the original requirement